Game/space-invaders/src/players/PlayerManager.cpp

```diff
@@ -0,0 +1,115 @@
+﻿#include "PlayerManager.h"
+
+#include <GLFW/glfw3.h>
+
+#include "../ResourceManager.h"
+#include "../utils/GameTime.h"
+
+PlayerManager::PlayerManager(IProjectileHandler& projectileHandler)
+    : m_ProjectileHandler(projectileHandler)
+{
+    
+}
+
+PlayerManager::~PlayerManager()
+{ }
+
+void PlayerManager::CreatePlayer(float screenWidth, float screenHeight)
+{
+    std::cout << "Creating player\n";
+    
+    const std::string playerTextureName = "Player";
+    const std::string playerTexturePath = "res/textures/ship.png";
+    
+    m_PlayerSprite = ResourceManager::LoadTexture(playerTexturePath, playerTextureName, true);
+    
+    const glm::vec2 playerInitialPosition = glm::vec2(
+        (screenWidth / 2.0f) - (PLAYER_SIZE.x / 2.0f),
+        screenHeight - PLAYER_SIZE.y - PLAYER_OFFSET.y);
+    
+    m_Player = std::make_shared<GameObject>(
+        playerInitialPosition,
+        PLAYER_SIZE,
+        m_PlayerSprite,
+        PLAYER_COLOR,
+        glm::vec2(PLAYER_SPEED)
+        );
+}
+
+void PlayerManager::ProcessInput(float deltaTime, const Input& input, float boundsWidth)
+{
+    const float velocity = PLAYER_SPEED * deltaTime;
+
+    if (input.GetKey(GLFW_KEY_A))
+    {
+        if (m_Player->Position.x >= 0 + PLAYER_OFFSET.x)
+        {
+            m_Player->Position.x -= velocity; 
+        }
+    }
+    
+    else if (input.GetKey(GLFW_KEY_D))
+    {
+        if (m_Player->Position.x <= boundsWidth - (m_Player->Size.x + PLAYER_OFFSET.x))
+        {
+            m_Player->Position.x += velocity; 
+        }
+    }
+
+    if (input.GetKey(GLFW_KEY_SPACE))
+    {
+        if (CheckIfCanShootAgain())
+        {
+            std::cout << "SHOT!!\n";
+            
+            CreateProjectile();
+            
+            m_LastShotTime = GameTime::Time;
+            bCanShoot = false;
+        }
+    }
+}
+
+void PlayerManager::CreateProjectile()
+{
+    // load sprite
+    const std::shared_ptr<Texture> projectileSprite = ResourceManager::LoadTexture(
```

Comment:
This would cause to create a texture each time the projectile is spawned. Use load just once, like on PlayerManager constructor and then here use the GetTexture method. Or on constructor, use load and hold a member shared_ptr to the texture and reuse it here.